src/java/map/ConcurrentBinaryMask.java

```diff
@@ -33,90 +35,150 @@ public ConcurrentBinaryMask(ConcurrentBinaryMask mask, long seed, String name) {
 	}
 
 	public ConcurrentBinaryMask randomize(float density) {
+		if (VisualDebugger.ENABLED) {
+			binaryMask.randomize(density);
+			return this;
+		}
```

Comment:
What if instead of duplicating this a lot of times this call get's moved to the Pipeline.add() function?

Reply:
see general comment on the PR